Libraries/xcassets/Headers/xcassets/Asset/Asset.h

```diff
@@ -29,6 +29,7 @@ class Asset {
 private:
     FullyQualifiedName         _name;
     std::string                _path;
+    libutil::Filesystem const *_filesystem;
```

Comment:
the filesystem object is passed to run(), run() creates the catalog, and passes that to CompileAction() and does not return it, and would not use it past the write() call at the end of run(). I don't see why I can't expect the filesystem to have the same lifetime as the Catalog object.